common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/ClientCertAuthChallengeHandler.java

```diff
@@ -406,22 +407,24 @@ private SmartcardPinDialog.PositiveButtonListener getSmartcardPinDialogPositiveB
         return new SmartcardPinDialog.PositiveButtonListener() {
             @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
             @Override
-            public void onClick(@NonNull final String pin) {
+            public void onClick(@NonNull final char[] pin) {
                 synchronized (sDeviceLock) {
                     // Need to request a PivSession in order to access certs on YubiKey.
                     getActivePivSessionAsync(request, new IPivSessionCallback() {
                         @Override
                         public void onGetSession(@NonNull PivSession piv) {
                             try {
                                 //Verify PIN and handle results
-                                verifySmartcardPin(pin.toCharArray(), certDetails, request, piv);
+                                verifySmartcardPin(pin, certDetails, request, piv);
                             } catch (final IOException | ApduException | BadResponseException e) {
                                 Logger.error(methodTag, e.getMessage(), e);
                                 //Show general error dialog.
                                 mDialogHolder.showErrorDialog(R.string.smartcard_general_error_dialog_title, R.string.smartcard_general_error_dialog_message);
                                 //Emit telemetry for this exception.
                                 Telemetry.emit(new ErrorEvent().putException(e));
                                 request.cancel();
+                                //Clear PIN attempt from local char array.
+                                clearPin(pin);
```

Comment:
put this in the final {} block here, and you won't have to put it anywhere else.